File: src/1324-Print-Words-Vertically/1324.java
class Solution {
    public List<String> printVertically(String s) {
        String[] strs = s.split(" ");
        int max_len = 0;
        for (String st : strs) max_len = Math.max(max_len, st.length());
        
        List<String> res = new ArrayList();
        for (int j = 0; j < max_len; j++) {
            StringBuilder t = new StringBuilder();
            int t_len = 0;
            
            for (String st : strs) {
                if (st.length() > j) {
                    t.append(st.charAt(j));
                    t_len = t.length();
                } else t.append(" ");
            }
            t.setLength(t_len);
            res.add(t.toString());
        }
        return res;
    }
}

File: src/1324-Print-Words-Vertically/1324.js
var printVertically = function(s) {
    let strs = s.split(" "), max_len = 0;
    for (let st of strs) {
        max_len = Math.max(max_len, st.length);
    }
    
    let res = [];
    for (let j = 0; j < max_len; j++) {
        let t = "", t_len = 0;
        for (let st of strs) {
            if (st.length > j) {
                t += st[j];
                t_len = t.length;
            } else t += " ";
        }
        res.push(t.substring(0, t_len));
    }
    return res;
};